[FIX] Non-files showing up in auto file context feature

charlwillia6 opened the pull request:
## Description ✏️

Fixes issue with clicking inside other tabs in the code editor, causing non filenames to show up in the Current File feature in the chat input.  

This has only had a little bit of testing.

## Checklist ✅

- [ ] I have added screenshots (if UI changes are present).
- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main' into charlwillia6/fix-output-as-file-in-auto-file-context-feature

charlwillia6 committed:
Fixed issue with current file showing invalid text

charlwillia6 committed:
Delete chat file. Added it to git ignore.

charlwillia6 committed:
[FEAT] Add session-based top bar and first prompt indicator

Introduces a new `isFirstInSession` prop to the `ContinueInputBox` and `TipTapEditor` components. This prop determines whether the top bar should be displayed for the first input in a session. The top bar is now only shown for the first input in a session, providing a cleaner interface. Additionally, the `ContinueInputBox` now renders a visual indicator to indicate when the user is entering the first prompt in a new session.

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main' into charlwillia6/fix-output-as-file-in-auto-file-context-feature

charlwillia6 committed:
Fix: prevent invalid text in ActiveFileIndicator during new session

- Add file path validation in uiStateSlice's setActiveFilePath reducer
- Add validation check in VsCodeExtension's updateNewWindowActiveFilePath
- Remove redundant activeFilePath management from stateSlice
- Fix import path for isValidFilePath utility

This fixes an issue where text from the Output console could appear in
the ActiveFileIndicator when clicking New Session by ensuring file paths
are validated at all entry points and managed in a single location.

charlwillia6 committed:
Update .gitignore

Fryingpannn commented on extensions/vscode/src/extension/VsCodeExtension.ts:
```diff
@@ -375,12 +376,18 @@ export class VsCodeExtension {
+        if (!event) return;
+        const filepath = event.document.uri.fsPath;
+        if (isValidFilePath(filepath)) {
+          this.core.invoke("didChangeActiveTextEditor", { filepath });
```
Does the invoke need to be within the valid file path? it wasn't originally. maybe only our request setActiveFilePath needs to be within

charlwillia6 replied:
Yes, I think this was an extra check. I can revert it back and test. We may not need it.

Fryingpannn commented on gui/src/components/mainInput/TipTapEditor.tsx:
```diff
@@ -970,7 +972,7 @@ const TipTapEditor = memo(function TipTapEditor({
       }}
     > 
-      {(!isPerplexity && !isAider) && <TopBar />}
+      {(!isPerplexity && !isAider) && isFirstInSession && <TopBar />}
```
instead of simply not displaying topbar, we should modify the topbar itself, and make the current file sort of greyed out and unable to be X'd. Should just always display the topbar, just modify how it looks within.

we may have other buttons to be put in the top bar in future. should not be disappeared.

charlwillia6 replied:
Okay, I will do that then and update the PR.

Fryingpannn commented on gui/src/pages/gui.tsx:
```diff
@@ -356,6 +356,7 @@ function GUI() {
                         isMainInput={false}
                         editorState={item.editorState}
                         contextItems={item.contextItems}
+                        isFirstInSession={index === state.history.findIndex(item => item.message.role === "user")}
```
I think this index value can be `useMemo`'d before

Fryingpannn reviewed (changes requested):
Seems like although on the UI it's not indicating the wrong file anymore, though underlying it's still picking up the wrong thing. I clicked on output tab then pressed enter. Under context we can still see the active file is "tasks" which is what you get by pressing the output console tab.
<img width="500" alt="image" src="https://github.com/user-attachments/assets/452f6516-68e2-4dc7-b714-d3259e594633">

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main' into charlwillia6/fix-output-as-file-in-auto-file-context-feature

charlwillia6 committed:
Refactored the VSCode extension and GUI components to improve active file handling. Key changes include:

- Updated VsCodeExtension.ts to streamline active file path management.
- Enhanced VsCodeIde in ideProtocol.ts with methods for setting and retrieving the active file path.
- Modified ActiveFileIndicator.tsx and TopBarIndicators.tsx to conditionally display active file information based on session state.
- Simplified state management logic in uiStateSlice.ts.

charlwillia6 committed:
perf: Memoize first session message check in GUI

charlwillia6 commented:
@Fryingpannn PR has been updated.  All comments have been resolved.

Fryingpannn commented on extensions/vscode/src/extension/VsCodeExtension.ts:
```diff
@@ -371,16 +372,23 @@ export class VsCodeExtension {
     });
 
-    this.ide.onDidChangeActiveTextEditor((filepath) => {
-      this.core.invoke("didChangeActiveTextEditor", { filepath });
```
why is invoke removed?

Fryingpannn commented on extensions/vscode/src/extension/VsCodeExtension.ts:
```diff
@@ -371,16 +372,23 @@ export class VsCodeExtension {
-      this.sidebar.webviewProtocol.request("setActiveFilePath", filepath, [PEAR_CONTINUE_VIEW_ID]);
-    });
+    context.subscriptions.push(
+      vscode.window.onDidChangeActiveTextEditor((event) => {
```
did we have to change to use the vscode's onDidChangeActiveTextEditor? Is it just to have the `!event return`?

Fryingpannn commented on gui/src/pages/gui.tsx:
```diff
@@ -175,6 +176,10 @@ function GUI() {
   const state = useSelector((state: RootState) => state.state);
 
+  const getIsFirstInSession = useMemo(() => {
+    return (idx: number) => idx === state.history.findIndex(item => item.message.role === "user");
```
why can't just check history.length?

Fryingpannn replied:
const isFirstMessage = state.history.length === 0;

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main' into charlwillia6/fix-output-as-file-in-auto-file-context-feature

Fryingpannn commented on gui/src/components/mainInput/ActiveFileIndicator.tsx:
```diff
@@ -5,7 +5,11 @@ import { useCallback, useContext, useEffect } from "react";
 
-export default function ActiveFileIndicator() {
+interface ActiveFileIndicatorProps {
+  isFirstInSession?: boolean;
```
I think instead of using prop. We can just check the history within active file indicator. See this PR: https://github.com/trypear/pearai-submodule/pull/181/files#diff-ea630c971a646645666575d7bbfe531ea6c00a5cb7108f3dc66ded51fccfd963R20

This PR I made above i will not be merging. Was playing around with some stuff but not good. However I did do the history stuff in there to determine first chat input, which may be worth using instead of prop drilling. Also we should grey out the thing, with unchangeable text if conversation is prompted.

charlwillia6 commented:
PR on hold. Currently polling the community as to whether the current file feature should be removed or not.

charlwillia6 commented:
Active FIle feature being removed and chat interface is being redesigned.  Closing this PR.
